x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/last_value.tsx

```diff
@@ -0,0 +1,217 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiSelect, EuiButtonGroup } from '@elastic/eui';
+import { OperationDefinition } from './index';
+import { FieldBasedIndexPatternColumn } from './column_types';
+import { IndexPatternField, IndexPattern } from '../../types';
+import { updateColumnParam } from '../layer_helpers';
+import { DataType } from '../../../types';
+
+function ofName(name: string) {
+  return i18n.translate('xpack.lens.indexPattern.lastValueOf', {
+    defaultMessage: 'Last value of {name}',
+    values: { name },
+  });
+}
+
+const supportedTypes = new Set(['string', 'boolean', 'number', 'ip']);
+
+function getDateFields(indexPattern: IndexPattern): IndexPatternField[] {
+  const dateFields = indexPattern.fields.filter((field) => field.type === 'date');
+  if (indexPattern.timeFieldName) {
+    dateFields.sort(({ name: nameA }, { name: nameB }) => {
+      if (nameA === indexPattern.timeFieldName) {
+        return -1;
+      }
+      if (nameB === indexPattern.timeFieldName) {
+        return 1;
+      }
+      return 0;
+    });
+  }
+  return dateFields;
+}
+
+export interface LastValueIndexPatternColumn extends FieldBasedIndexPatternColumn {
+  operationType: 'last_value';
+  params: {
+    sortField: string;
+    sortOrder: 'asc' | 'desc';
+    // last value on numeric fields can be formatted
+    format?: {
+      id: string;
+      params?: {
+        decimals: number;
+      };
+    };
+  };
+}
+
+export const lastValueOperation: OperationDefinition<LastValueIndexPatternColumn, 'field'> = {
+  type: 'last_value',
+  displayName: i18n.translate('xpack.lens.indexPattern.lastValue', {
+    defaultMessage: 'Last value',
+  }),
+  input: 'field',
+  onFieldChange: (oldColumn, field) => {
+    const newParams = { ...oldColumn.params };
+
+    if ('format' in newParams && field.type !== 'number') {
+      delete newParams.format;
+    }
+    return {
+      ...oldColumn,
+      dataType: field.type as DataType,
+      label: ofName(field.displayName),
+      sourceField: field.name,
+      params: newParams,
+    };
+  },
+  getPossibleOperationForField: ({ aggregationRestrictions, type }) => {
+    if (supportedTypes.has(type) && !aggregationRestrictions) {
+      return { dataType: type as DataType, isBucketed: false, scale: 'ordinal' };
+    }
+  },
+  getDisabledStatus(indexPattern: IndexPattern) {
+    const hasDateFields = indexPattern && getDateFields(indexPattern).length;
+    if (!hasDateFields) {
+      return i18n.translate('xpack.lens.indexPattern.lastValue.disabled', {
+        defaultMessage: 'This function requires the presence of a date field in your index',
+      });
+    }
+  },
+  buildColumn({ field, previousColumn, indexPattern }) {
+    const sortField =
+      indexPattern.timeFieldName ||
+      indexPattern.fields.find((f) => f.type === 'date')?.name ||
+      field.name;
```

Comment:
Why using the selected field here? As this might be a non-date field in some cases I'm not sure it would do the right thing. If I read the code correctly, this situation should never happen, so it should be fine to throw an exception in that case.